extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/restclient/OpenTelemetryClientFilter.java

```diff
@@ -130,17 +130,19 @@ public void set(final ClientRequestContext carrier, final String key, final Stri
     private static class ClientSpanNameExtractor implements SpanNameExtractor<ClientRequestContext> {
         @Override
         public String extract(final ClientRequestContext request) {
-            String pathTemplate = (String) request.getProperty("UrlPathTemplate");
-            if (pathTemplate != null && pathTemplate.length() > 1) {
-                return pathTemplate;
+            StringBuilder name = new StringBuilder("HTTP");
+
+            String method = request.getMethod();
+            if (method != null && method.length() > 1) {
+                name.append(" ").append(method);
             }
 
-            String uriPath = request.getUri().getPath();
-            if (uriPath != null && uriPath.length() > 1) {
-                return uriPath;
+            String pathTemplate = (String) request.getProperty("UrlPathTemplate");
+            if (pathTemplate != null && pathTemplate.length() > 1) {
+                name.append(" ").append(pathTemplate);
```

Comment:
I'm unsure if I agree with merging the method name with the route for the span name. I agree that we were not spec compliant with the URI, but this does not seem to be compliant either. 

In my opinion, we should just use the behavior from the OTel `SpanNameExtractor`. You can disambiguate the client and server spans with the `span.kind` tag.

Reply:
If you look into the entire sentence about Client Spans:

>  However, the same approach usually does not work for HTTP client spans, especially when instrumentation is provided by a lower-level middleware that is not aware of the specifics of how the URIs are formed. Therefore, HTTP client spans SHOULD be using conservative, low cardinality names formed from the available parameters of an HTTP request, such as "HTTP {METHOD_NAME}". Instrumentation MUST NOT default to using URI path as span name, but MAY provide hooks to allow custom logic to override the default span name.

From my interpretation, the name should either be "HTTP {METHOD_NAME}" or not the URI, but it is unclear which other names can be used. It leaves it open to provide hooks to override the name (which we don't have at the moment).

Their implementation does not use the Route name for the Client Span, but I think it is reasonable to use it because it is already used for the Server Span, and for most HTTP Clients, there is no route available.

If we go with your approach and we mix Quarkus OpenTelemetry with the OpenTelemetry Agent, we end up in a situation where Quarkus HTTP Clients generate the Clients Spans with ´HTTP {METHOD_NAME} Route´, and the ones handled by the Agent will generate ´HTTP {METHOD_NAME}´ only, which in my opinion is more confusing.

Yes, specifications should give you freedom, but must also standardize things. It would also be frustrating if each platform generates the name differently, and you end up with all of these in your final trace. I think it would be reasonable to reach out to the OTel specification / Java implementation and do this change directly there, so everyone can benefit from it, and hopefully, the name to be consistent across everybody.

Reply:
Using stock `HttpSpanNameExtractor` instead.